src/components/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,263 @@
+    <li>
+      <Link
+        key={id}
+        id={id}
+        type="action"
+        href={href}
+        isUnderlined={false}
+        screenreaderOnlyText={screenreaderOnlyText}
+        aria-current={isSelected ? "page" : null}
+        className={isSelected ? "selectedItem" : ""}
+        sx={childrenStyles.outLine}
+      >
+        {children}
+      </Link>
+    </li>
+  );
+};
+
+/**
+ * The `SubNav` component is a navigation element that displays a group of
+ * related action items (buttons or links) in a horizontal layout.  The action
+ * items in the `SubNav` component will link to children within or perform
+ * actions related to the current section of a website.
+ */
+export const SubNav: ChakraComponent<
+  React.ForwardRefExoticComponent<
+    React.PropsWithChildren<SubNavProps> & React.RefAttributes<HTMLDivElement>
+  >,
+  SubNavProps
+> = chakra(
+  forwardRef<HTMLDivElement, React.PropsWithChildren<SubNavProps>>(
+    (props, _ref?) => {
+      const {
+        className,
+        actionBackgroundColor,
+        highlightColor,
+        primaryActions,
+        secondaryActions,
+      } = props;
+
+      const { scrollableRef, showRightFade } = useScrollFadeStyles();
+
+      if (actionBackgroundColor !== undefined && highlightColor === undefined) {
+        console.warn(
+          "NYPL Reservoir SubNav: The `actionBackgroundColor` prop has been passed, but the `highlightColor` prop has not been passed. Because of this, the `actionBackgroundColor` prop will be ignored."
+        );
+      }
+
+      const validateActions = (actions: React.ReactNode, propName: string) => {
+        if (React.isValidElement(actions)) {
+          React.Children.forEach(
+            actions.props.children,
+            (child: React.ReactElement) => {
+              if (child.type !== SubNavButton && child.type !== SubNavLink) {
+                console.warn(
+                  `NYPL Reservoir SubNav: An element that is not a SubNavButton or SubNavLink component has been passed in the \`${propName}\` prop. That element may not work properly.`
+                );
+                return null;
+              }
+            }
+          );
+        }
+      };
+
+      // Validate primaryActions and secondaryActions
+      validateActions(primaryActions, "primaryActions");
+      validateActions(secondaryActions, "secondaryActions");
+
+      const backgroundColor =
+        highlightColor !== undefined ? actionBackgroundColor : undefined;
+
+      const styles = useMultiStyleConfig("SubNav", {
+        backgroundColor: backgroundColor,
+        highlightColor: highlightColor,
+      });
+
+      return (
+        <Box as="nav" aria-label="Sub-navigation menu" __css={styles.base}>
+          <Box __css={styles.container}>
+            <Flex
+              alignItems="center"
+              className={className}
+              gap="1rem"
+              justify="space-between"
+            >
+            <div style={styles.primaryList}>
```

Comment:
I think this is throwing a linting or typescript error. Convert the div into a `Box`, and you can also replace `style` with `sx`.